app/views/devise/registrations/edit.html.erb

```diff
@@ -0,0 +1,47 @@
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
+<%= render partial: "shared/flash"%>
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field my-2">
+  <%= f.text_field :name, autofocus: true, class: "form-control", placeholder: "Full name" %>
+</div>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+    <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
+  <% end %>
```

Comment:
hi @chrystalme  I agree with you,  In general, it is a good practice to put as much logic as possible in the controller or model instead of the view. [This makes it easier to test your code and also makes it more maintainable ](https://stackoverflow.com/questions/3024385/how-add-logic-to-views-ruby-on-rails)[1](https://stackoverflow.com/questions/3024385/how-add-logic-to-views-ruby-on-rails).

However, there are some cases where it might be necessary to use logic in views. [For example, if you need to display different content based on the user’s role or if you need to display different content based on the current date](https://stackoverflow.com/questions/1216766/how-much-logic-do-you-put-in-views) And to be more specific  didn't add any change, on device generated it for me.

Reply:
In these cases, you use the helper to encapsulate the logic and call the method in the view

Reply:
Bro, why I can do that? since we are using the device gem to generate authentication,  I don't see any issues with this!  And if you believe it necessary please help me with how can I implement it(give me away)!